### PR TITLE
Feature/track 5 switch to postgres

### PR DESCRIPTION
**Pull request by j16r**
switching over to postgres y'all

done:

 - [x] password needs to be added to production.ini
 - [x] groups need to be added to production.ini
 - [x] travis build :/
 - [x] need to put db in volume
 - [x] letsencrypt on initial deploy is broken

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -15,5 +19,6 @@ notifications:
     on_start: never     # options: [always|never|change] default: always
 
 addons:
-    code_climate:
-        repo_token: 5db3f72ce9253ebdff56c2f98793691557a46415e09f70ab9c46036216d22580
+  postgresql: "9.5"
```
Comment (tomca32) [inline .travis.yml]: Why 9.5 and not 9.6?

File: deploy/README.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
+# Deployment
```
Comment (tomca32) [inline deploy/README.md]: Good call on documenting the deploy process.

File: deploy/README.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+Since this is an open source project and we'd like to keep everything in one
+place, we keep inventories encrypted. For vault passwords you'll need to speak
+to one of the maintainers.
+
+## Roles
+
+ * _**server**_ - Server role is used on all servers, depends on ansible galaxy
+   role `kamaln7.swapfile`.
+ * _**db**_ - Database role, depends on ansible galaxy role
```
Comment (tomca32) [inline deploy/README.md]: db role also depends on db_volume role, no?

File: deploy/roles/webserver/tasks/main.yml [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+    - libyaml-dev
+    - nginx-extras
+    - zlib1g-dev
+    - nodejs
+
+  # LetsEncrypt setup, these must happen in order and should only happen on
+  # very first deploy
+- name: Check if SSL cert has been installed
+  shell: "test ! -e {{ ssl_certificate }}"
+  register: obtain_letsencrypt_cert
```
Comment (tomca32) [inline deploy/roles/webserver/tasks/main.yml]: This is an elegant solution, wasn't aware this was possible with Ansible. Would it make sense to move all SSL tasks into their own role?

File: deploy/roles/db/meta/main.yml [hunk 5]
```diff
@@ -0,0 +1,10 @@
+---
+dependencies:
+  - db_volume
+  - role: geerlingguy.ansible-role-postgresql
+    postgresql_data_dir: /mnt/data/db
+    postgresql_databases:
+    - name: tracker
```
Comment (tomca32) [inline deploy/roles/db/meta/main.yml]: Could we change this to mindleaps_tracker? Total nitpick but I like to be more specific.

File: deploy/roles/db/meta/main.yml [hunk 6]
```diff
@@ -0,0 +1,10 @@
...
+    postgresql_users:
+    - name: tracker
```
Comment (tomca32) [inline deploy/roles/db/meta/main.yml]: Same as ^^

File: deploy/roles/webserver/templates/database.yml.j2 [hunk 7]
```diff
@@ -0,0 +1,8 @@
+production:
+  adapter: postgresql
+  encoding: unicode
+  host: {{ database_address }}
+  database: tracker
```
Comment (tomca32) [inline deploy/roles/webserver/templates/database.yml.j2]: Maybe change db name to mindleaps_tracker?

File: deploy/roles/webserver/templates/database.yml.j2 [hunk 8]
```diff
@@ -0,0 +1,8 @@
...
+  pool: 5
+  username: tracker
```
Comment (tomca32) [inline deploy/roles/webserver/templates/database.yml.j2]: ^^

**Comment by tomca32** [comment]
I added some specific comments and questions but generally I think all of this makes sense.

File: deploy/roles/webserver/tasks/main.yml (hunk 4 above)
Comment (j16r) [inline deploy/roles/webserver/tasks/main.yml]: The roles would be very tightly coupled and the ordering would be less obvious so I don't think it'd be worthwhile.

**Comment by tomca32** [comment]
Great work on this. Shall we merge it?

**Pull request merged**